#$ -S /bin/bash
#$ -q rnd.q,test.q,prod.q
#$ -cwd
#$ -V
#$ -p -1000

set

PROJECT=$1
SM_TAG=$2
REF_GENOME=$3
DBSNP=$4
DNA_HASH_ADDRESS=$5
MDNA_HASH_ADDRESS=$6

RIS_ID=${SM_TAG%@*}
BARCODE_2D=${SM_TAG#*@}

JAVA_1_7="/isilon/sequencing/Kurt/Programs/Java/jdk1.7.0_25/bin"
CORE_PATH="/isilon/sequencing/Seq_Proj/"
BWA_DIR="/isilon/sequencing/Kurt/Programs/BWA/bwa-0.7.8/"
PICARD_DIR="/isilon/sequencing/Kurt/Programs/Picard/picard-tools-1.118"
GATK_DIR="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_3/GenomeAnalysisTK-3.1-1"
VERIFY_DIR="/isilon/sequencing/Kurt/Programs/VerifyBamID/verifyBamID_20120620/bin/"
BED_DIR=$CORE_PATH"/BED_Files/"
GENE_LIST="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/RefSeqGene.GRCh37.Ready.txt"
VERIFY_VCF="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/Omni25_genotypes_1525_samples_v2.b37.PASS.ALL.sites.vcf"
CODING_BED="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/UCSC_hg19_CodingOnly_083013_MERGED_noContigs_noCHR.bed"
CODING_BED_MT="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/MT.coding.bed"
TRANSCRIPT_BED="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/Transcripts.UCSC.Merged.NoContigsAlts.bed"
TRANSCRIPT_BED_MT="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/MT.transcripts.bed"
GAP_BED="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/GRCh37.gaps.bed"
SAMTOOLS_DIR="/isilon/sequencing/Kurt/Programs/samtools/samtools-0.1.18/"
TABIX_DIR="/isilon/sequencing/Kurt/Programs/TABIX/tabix-0.2.6/"
LUMPY_DIR="/isilon/sequencing/Kurt/Programs/LUMPY/lumpy-sv-master"
KEY="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_2/lee.watkins_jhmi.edu.key"
SAMTOOLS_HTS=" /isilon/sequencing/peng/softwares/samtools-1.2/"

mkdir -p $CORE_PATH/$PROJECT/LOGS
mkdir -p $CORE_PATH/$PROJECT/BAM/{AGGREGATE,READGROUP}
mkdir -p $CORE_PATH/$PROJECT/HC_BAM
mkdir -p $CORE_PATH/$PROJECT/GVCF
mkdir -p $CORE_PATH/$PROJECT/REPORTS/{ALIGNMENT_SUMMARY,ANNOVAR,PICARD_DUPLICATES,QC_REPORTS,TI_TV,TI_TV_MS,VERIFY_BAM_ID,SAMPLE_SHEETS,OXIDATION,JUMPING,ESTIMATE_LIBRARY}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/TI_TV/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/TI_TV_MS/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/BASECALL_Q_SCORE_DISTRIBUTION/{METRICS,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/CONCORDANCE/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/CONCORDANCE_MS/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/COUNT_COVARIATES/{GATK_REPORT,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/GC_BIAS/{METRICS,PDF,SUMMARY}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/INSERT_SIZE/{METRICS,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/LOCAL_REALIGNMENT_INTERVALS
mkdir -p $CORE_PATH/$PROJECT/REPORTS/MEAN_QUALITY_BY_CYCLE/{METRICS,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/DEPTH_OF_COVERAGE/{DEPTH_SUMMARY,CODING_COVERAGE,TRANSCRIPT_COVERAGE}
mkdir -p $CORE_PATH/$PROJECT/TEMP
mkdir -p $CORE_PATH/$PROJECT/FASTQ
mkdir -p $CORE_PATH/$PROJECT/BED_Files
mkdir -p $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/PASS
mkdir -p $CORE_PATH/$PROJECT/VCF/SINGLE/CODING/PASS
mkdir -p $CORE_PATH/$PROJECT/VCF/MULTI/WHOLE_GENOME/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/VCF/MULTI/CODING/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/SNV/SINGLE/WHOLE_GENOME/PASS
mkdir -p $CORE_PATH/$PROJECT/SNV/SINGLE/CODING/PASS
mkdir -p $CORE_PATH/$PROJECT/SNV/MULTI/WHOLE_GENOME/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/SNV/MULTI/CODING/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/INDEL/SINGLE/WHOLE_GENOME/PASS
mkdir -p $CORE_PATH/$PROJECT/INDEL/SINGLE/CODING/PASS
mkdir -p $CORE_PATH/$PROJECT/INDEL/MULTI/WHOLE_GENOME/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/INDEL/MULTI/CODING/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/LUMPY/{RAW,UCSC_CODING}

# From the unaligned bam file, remove duplicated data (have to sort by query name)

($SAMTOOLS_HTS/samtools view -H $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/Alignment/Unmapped/alignments.bam ; \
$SAMTOOLS_HTS/samtools view $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/Alignment/Unmapped/alignments.bam \
| sort -k 1 \
| uniq) \
| $SAMTOOLS_HTS/samtools view -1hS - \
| $SAMTOOLS_HTS/samtools sort -m 4G -@ 4 \
-T $CORE_PATH/$PROJECT/TEMP \
-o $CORE_PATH/$PROJECT/TEMP/$SM_TAG".unmapped.bam" -

## --Merge Aligned and Unaligned reads--

$JAVA_1_7/java -jar $PICARD_DIR/MergeSamFiles.jar \
INPUT=$CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/Recalibration/alignments.bam \
INPUT=$CORE_PATH/$PROJECT/TEMP/$SM_TAG".unmapped.bam" \
USE_THREADING=TRUE \
CREATE_INDEX=TRUE \
CREATE_MD5_FILE=TRUE \
OUTPUT=$CORE_PATH/$PROJECT/BAM/$SM_TAG".bam"

# Nevermind below. Can't frigging index the output

# $SAMTOOLS_HTS/samtools cat \
# $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/Recalibration/alignments.bam \
# $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/Alignment/Unmapped/alignments.bam \
# >| $CORE_PATH/$PROJECT/BAM/$SM_TAG".bam"
# 
# $SAMTOOLS_HTS/samtools index \
# $CORE_PATH/$PROJECT/BAM/$SM_TAG".bam"

# delete bina's bam files.

# rm -rvf $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/Recalibration/alignments.bam*
# rm -rvf $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/Alignment/Unmapped/alignments.bam*


qsub -N "A.01_DOC_AUTO_CODING."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.AUTO.CODING.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.AUTO.CODING.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.01_DOC_AUTO_CODING.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.02_DOC_AUTO_WG."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.AUTO.WG.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.AUTO.WG.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.02_DOC_AUTO_WG.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.03_DOC_CODING."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.CODING.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.CODING.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.03_DOC_CODING.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.04_DOC_TRANSCRIPT."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.TRANSCRIPT.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".DOC.TRANSCRIPT.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.04_DOC_TRANSCRIPT.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.05_VERIFY_BAM_ID."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".VERIFY.BAM.ID.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".VERIFY.BAM.ID.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.05_VERIFY_BAM_ID.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.06_INSERT_SIZE."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".INSERT.SIZE.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".INSERT.SIZE.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.06_INSERT_SIZE.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.07_ALIGNMENT_SUMMARY_METRICS."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".ALIGNMENT.SUMMARY.METRICS.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".ALIGNMENT.SUMMARY.METRICS.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.07_ALIGNMENT_SUMMARY_METRICS.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.08_BASECALL_Q_SCORE_DISTRIBUTION."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".BASECALL.Q.SCORE.DISTRIBUTION.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".BASECALL.Q.SCORE.DISTRIBUTION.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.08_BASECALL_Q_SCORE_DISTRIBUTION.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.09_GC_BIAS."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".GC.BIAS.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".GC.BIAS.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.09_GC_BIAS.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.10_MEAN_QUALITY_BY_CYCLE."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".MEAN.QUALITY.BY.CYCLE.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".MEAN.QUALITY.BY.CYCLE.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.10_MEAN_QUALITY_BY_CYCLE.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.11_OXIDATION."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".OXIDATION.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".OXIDATION.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.11_OXIDATION.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "A.13_ESTIMATE_LIBRARY."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".ESTIMATE.LIBRARY.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".ESTIMATE.LIBRARY.log" \
$CORE_PATH/$PROJECT/SCRIPTS/A.13_ESTIMATE_LIBRARY.sh $1 $2 $3 $4 $5 $6

sleep 3s

qsub -N "BAM2FQ."$RIS_ID"_"$BARCODE_2D \
-o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".BAM2FASTQ.log" \
-e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".BAM2FASTQ.log" \
/isilon/sequencing/Kurt/NICE_SCRIPTS/BAM2FQ.AGGREGATE.SGE.sh $CORE_PATH/$PROJECT/BAM/$SM_TAG".bam" $CORE_PATH/$PROJECT/FASTQ

# sleep 3s
# 
# qsub -N "B.01_ANNOTATE_VCF."$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".ANNOTATE.VCF.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".ANNOTATE.VCF.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/"B.01_ANNOTATE_VCF.sh $1 $2 $3 $4 $5 $6
# 
# sleep 3s
# 
# qsub -N "B.02_VQSR_SNP_PLOT."$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".VQSR.SNP.PLOT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".VQSR.SNP.PLOT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.02_VQSR_SNP_PLOT.sh $1 $2 $3 $4 $5 $6
# 
# sleep 3s
# 
# qsub -N "B.03_VQSR_INDEL_PLOT."$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".VQSR.INDEL.PLOT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".VQSR.INDEL.PLOT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.03_VQSR_INDEL_PLOT.sh $1 $2 $3 $4 $5 $6
